### PR TITLE
Refactor: move modules to new domain-rrd folder

### PR DESCRIPTION
**Pull request by yoomlam**
Refactor code to comply with [Software Conventions](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Software-Conventions).
RRD (Rapid Ready for Decision) features implemented in v1 and v2 should be extracted in a domain folder that is separate from VRO "platform" code.

This PR is a first step: moves Java folders and renames some packages to be under the `...rrd` package. Subsequent PRs will make further refinements to keep each PR as small as possible.

[Slack](https://dsva.slack.com/archives/C01Q7979Z7D/p1679519446850369?thread_ts=1678207065.568629&cid=C01Q7979Z7D)

### CODE REVIEW

File: service/spi/build.gradle
Comment (yoomlam) [inline service/spi/build.gradle:1]: This file was moved to https://github.com/department-of-veterans-affairs/abd-vro/pull/1330/files#diff-6839efb22a10f3eece934be194fadebc7dac457b2b9d3e0d64ec8f33a7e29875, not deleted

File: service/spi/src/main/java/gov/va/vro/service/spi/services/ClaimMetricsService.java
Comment (yoomlam) [inline service/spi/src/main/java/gov/va/vro/service/spi/services/ClaimMetricsService.java:1]: This file was moved to https://github.com/department-of-veterans-affairs/abd-vro/pull/1330/files#diff-6839efb22a10f3eece934be194fadebc7dac457b2b9d3e0d64ec8f33a7e29875, not deleted

**Comment by au2** [comment]
@yoomlam  [Software Conventions](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Software-Conventions) is good. I do not think I disagree with anything. Consider moving domain specific db table handling to domains as well. Postgres makes this easier with schemas and flyway migrations are schema specific I believe so each team maintains its own schema and migrations.

Can you please address the conflicts.

**Comment by yoomlam** [comment]
[PR: End-to-end test for v2](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4734107476) passed

**Comment by yoomlam** [comment]
@msnwatson Can you double-check this PR to ensure I'm not deleting any files unintentionally.
Files I intend to delete are:
- `api/build.gradle`: merged into `domain-rrd/rrd-api-controller/build.gradle`
- `service/spi/build.gradle`: merged into `domain-rrd/rrd-workflows/build.gradle`

File: domain-rrd/rrd-shared/src/main/java/gov/va/vro/service/spi/services/ClaimMetricsService.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package gov.va.vro.service.spi.services;
```
Comment (yoomlam) [inline domain-rrd/rrd-shared/src/main/java/gov/va/vro/service/spi/services/ClaimMetricsService.java:1]: This files was moved, not added as presented by GitHub.

**Review by msnwatson — APPROVED** [review]
Did not see any extra deleted files or any seeming changes in application logic as long as successful builds are completed (to verify proper dependency migration) and tests succeed (again to verify proper config with new structure)

**Comment by yoomlam** [comment]
[PR: End-to-end test for v2](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4736113711) failing inconsistently -- asked in [Slack](https://dsva.slack.com/archives/C01Q7979Z7D/p1681852075543809).

[Run](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4736848559/jobs/8409061929#step:8:4059) against [b5deac5](https://github.com/department-of-veterans-affairs/abd-vro/commit/b5deac5b6974bfc7a576a1ac26bd73399b7d508c)
```
Failed tests:
- gov.va.vro.end2end.VroV2Tests::testAutomatedClaimFullPositiveDuplicateBloodPressures()
- gov.va.vro.end2end.VroV2Tests::testAutomatedClaimFullPositivePresumptive()
```

**Comment by yoomlam** [comment]
[157240b](https://github.com/department-of-veterans-affairs/abd-vro/commit/157240beec0106221a6139a1384beb00f0b756fd) [failed](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4737077730/jobs/8409567847#step:8:3930) these tests:
```
Failed tests:
- gov.va.vro.end2end.VroV1Tests::positive7101Case01()
- gov.va.vro.end2end.VroV1Tests::positive6602Case01()
- gov.va.vro.end2end.VroV2Tests::testAutomatedClaimFullPositiveDuplicateBloodPressures()
- gov.va.vro.end2end.VroV2Tests::testAutomatedClaimFullPositiveIncompleteBloodPressures()
- gov.va.vro.end2end.VroV2Tests::testAutomatedClaimOrderExamNewClaim()
```

**Comment by yoomlam** [comment]
[PR: End-to-end test for v2](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4737784835)

**Pull request merged**